x-pack/plugins/fleet/server/services/elastic_agent_manifest.ts

```diff
@@ -68,6 +68,12 @@ spec:
                   fieldPath: metadata.name
             - name: STATE_PATH
               value: "/etc/elastic-agent"
+            - name: KUBERNETES_NODE_NAME
```

Comment:
Why is this env var needed when there is already the `NODE_NAME` which is exactly the same?